core/src/main/java/org/apache/spark/TaskContext.java

```diff
@@ -94,6 +95,20 @@ static void unset() {
   @Deprecated
   public abstract void addOnCompleteCallback(final Function0<Unit> f);
 
+  /**
+   * Add a listener to be executed when kill a task. We add this
+   * listener for some more clean works. For example, we need to register a "kill"
+   * callback to completely stop a receiver supervisor.
+   */
+  public abstract TaskContext addTaskKilledListener(TaskKilledListener listener);
+
+  /**
+   * Add a listener to be executed when kill a task. We add this
+   * listener for some more clean works. For example, we need to register a "kill"
+   * callback to completely stop a receiver supervisor.
+   */
+  public abstract TaskContext addTaskKilledCallback(final Function1<TaskContext, Unit> f);
```

Comment:
This should also be called `addTaskKilledListener` for consistency with the other API?